### ISSUE TITLE
GCP Kube cluster configuration to avoid deployment errors

### ISSUE THREAD

**Issue by kristianmandrup** [single-spa/import-map-deployer#29]
Tried to follow your guide and video tutorial to the best of my ability

When deploying image on GCP kube cluster: 

- Does not have minimum availability
- PodUnschedulable: Cannot schedule pods: Insufficient cpu.

I've written a detailed [step by step guide](https://github.com/kristianmandrup/mfe-root-config#kubernetes-cluster-setup-for-import-maps-deployer)

**Comment by joeldenning** [comment]
I don't know what those errors mean - "Insufficient CPU" sounds like you maybe have a budget set up in GCP that prevents adding the podf? And I don't know if you can lower the minimum availability or not.

I can't help you debug your GKE cluster, but wish you best of luck in getting the import-map-deployer running. The [pinned Youtube comment](https://www.youtube.com/watch?v=QHunH3MFPZs&lc=UgxlJBb1A_ekC8i-M694AaABAg) shows an alternative to GKE if you'd like to try that.

**Comment by kristianmandrup** [comment]
I resolve it. I pushed my luck adding `auto-scaling`, setting `min` and `max` both to 1. Not a good idea.

Anyways, I'm creating various Infrastructure as Code projects to make setting up this infra a bit more scalable. Currently experimenting with Google Cloud Run (managed containers) (as mentioned in the pinned Youtube comment) which looks a lot easier.

[gcr-terraform](https://github.com/kristianmandrup/gcr-terraform)

I've also started a couple of Pulumi projects:

- [pulumi-importmap-deployer-ts](https://github.com/kristianmandrup/pulumi-importmap-deployer-ts) sets up Google Storage bucket
- [pulumi-gcp-ts-gke](https://github.com/kristianmandrup/pulumi-gcp-ts-gke) sets up GCP kubernetes cluster with one node

These two projects need to be combined and include a policy for the cluster to have access to write to the bucket, obviously. Early days.

PS: In the video tutorials the font is very small.
